src/compiler/utilities.ts

```diff
@@ -4066,6 +4068,11 @@ export function isAssignmentTarget(node: Node): boolean {
     return getAssignmentTargetKind(node) !== AssignmentKind.None;
 }
 
+function isCompoundLikeAssignment(assignment: BinaryExpression): boolean {

```

Comment:
Since this is expected to only be used on definite assignments (i.e., `x = y`), it would be safer to restrict this with a more specific type.

```suggestion
function isCompoundLikeAssignment(assignment: AssignmentExpression<EqualsToken>): boolean {
```

I'd also suggest exporting this as `/** @internal */` so that it can be called from checker rather than changing `AssignmentKind`.